core/domain/rights_manager.py

```diff
@@ -633,6 +635,13 @@ def _change_activity_status(
         commit_cmds)
     _update_activity_summary(activity_type, activity_rights)
 
+    if new_status != ACTIVITY_STATUS_PRIVATE:
```

Comment:
Why is this separate from the if block on line 630, and why does line 639 duplicate line 631?